contracts/governance/Vesting/DevelopmentVesting.sol

```diff
@@ -0,0 +1,216 @@
+    event TokensWithdrawn(address indexed caller, address receiver, uint amount);
+    event TokensVested(address indexed caller, uint amount);
+    event VestedTokensWithdrawn(address indexed caller, address receiver, uint amount);
+
+    /**
+     * @dev Throws if called by any account other than the token owner or the contract owner.
+     */
+    modifier onlyOwners() {
+        require(msg.sender == tokenOwner || isOwner(), "unauthorized");
+        _;
+    }
+
+    /**
+     * @notice setup the vesting schedule
+     * @param _SOV the SOV token address
+     * @param _tokenOwner the owner of the tokens
+     * @param _cliff the cliff in seconds
+     * @param _duration the total duration in seconds
+     * @param _frequency the withdrawal interval in seconds
+     */
+    constructor(address _SOV, address _tokenOwner, uint _cliff, uint _duration, uint _frequency) public {
+        require(_SOV != address(0), "SOV address invalid");
+        require(_tokenOwner != address(0), "token owner address invalid");
+
+        _setSchedule(_cliff, _duration, _frequency);
+
+        SOV = IERC20(_SOV);
+        tokenOwner = _tokenOwner;
+    }
+
+    /**
+     * @notice set token owner
+     * @param _tokenOwner the owner of the tokens
+     */
+    function setTokenOwner(address _tokenOwner) public onlyOwner {
+        require(_tokenOwner != address(0), "token owner address invalid");
+        tokenOwner = _tokenOwner;
+    }
+
+    /**
+     * @notice send tokens to this contract, these tokens can be withdrawn any time by an owner
+     * @param _amount the amount of tokens to send
+     */
+    function sendTokens(uint _amount) public {
+        require(_amount > 0, "amount needs to be bigger than 0");
+
+        //retrieve the SOV tokens
+        bool success = SOV.transferFrom(msg.sender, address(this), _amount);
+        require(success);
+
+        amount += _amount;
+
+        emit TokensSent(msg.sender, _amount);
+    }
+
+    //TODO onlyOwners or onlyOwner ?
+    /**
+     * @notice withdraws tokens
+     * @param _amount the amount of tokens to be withdrawn
+     * @param _receiver the receiving address
+     */
+    function withdrawTokens(uint _amount, address _receiver) public onlyOwners {
+        require(_amount > 0, "amount needs to be bigger than 0");
+        require(_amount <= amount, "amount is not available");
+        require(_receiver != address(0), "receiver address invalid");
+
+        bool success = SOV.transfer(_receiver, _amount);
+        require(success);
+
+        amount -= _amount;
+
+        emit TokensWithdrawn(msg.sender, _receiver, _amount);
+    }
+
+    /**
+     * @notice change the vesting schedule
+     * @param _cliff the cliff in seconds
+     * @param _duration the total duration in seconds
+     * @param _frequency the withdrawal interval in seconds
+     */
+    function changeSchedule(uint _cliff, uint _duration, uint _frequency) public onlyOwner {
```

Comment:
if i set schedule and cliff to 0, will everything unlock immediately?
(that'd be good to have, because in the future we might want to be able to change the vesting schedule only from the current time. we'd need to be able to unlock all tokens and send them to another contract, which could be done in a single governance proposal)

Reply:
It was failed.
Changed, we can invoke `changeSchedule(0, 0, 0)` to unlock all tokens for all schedules. Test added.